WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt

```
@@ -596,7 +596,9 @@ class WooShippingLabelCreationViewModel @Inject constructor(
                     trackShippingRatesLoading(isSuccess = true)
                 } else {
                     updateState(ShippingRatesState.Error)
-                    trackShippingRatesLoading(isSuccess = false, error = shippingRatesResult.exceptionOrNull()?.message)
+                    val error = shippingRatesResult.exceptionOrNull()?.message
+                        ?: if (shippingRatesResult.getOrNull()?.isEmpty() == true) "no_rates_available" else null
+                    trackShippingRatesLoading(isSuccess = false, error = error)
```

np, just sharing a suggestion for a separate PR, not this one, I think we can improve this and also the `GetShippingRates` using the built-in helpers of the `Result` class:
- We can simplify `getShippingRates` logic using `Result#map`.
- We can simplify the logic here using `Result#fold`

I would also update `GetShippingRates` to return an error when the rates are empty, instead of having the logic here in the ViewModel, the `ViewModel` is already complex, and moving more logic to the Use cases would simplify it further.

---

These were good suggestions. I've updated `GetShippingRates` and `WooShippingLabelCreationViewModel`: https://github.com/woocommerce/woocommerce-android/pull/14323/commits/4216b383c1ec1197d4345e8694690b88fdee022d